### PR TITLE
refactor path to root preventing circular import

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Addressing potential issue in https://github.com/Lightning-AI/torchmetrics/pull/1886

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda @carmocca @justusschock @awaelchli @tchaton

### CODE REVIEW

File: .github/workflows/ci-tests-store.yml [hunk 1]
```diff
@@ -10,6 +10,11 @@ on:
     paths:
       - ".actions/*"
       - "requirements/store/**"
+      - "src/lightning/__about__.py"
+      - "src/lightning/__init__.py"
+      - "src/lightning/__main__.py"
+      - "src/lightning/__setup__.py"
+      - "src/lightning/__version__.py"
```
Comment (carmocca) [inline .github/workflows/ci-tests-store.yml:17]: These changes should be reflected in checkgroup

File: src/lightning/__init__.py [hunk 2]
```diff
@@ -29,9 +29,10 @@
 from lightning.pytorch.trainer import Trainer  # noqa: E402
 
 import lightning.app  # isort: skip # noqa: E402
-import lightning.store  # isort: skip # noqa: E402
+import lightning.store  # isort: skip # noqa: E402, F401
 
-lightning.app._PROJECT_ROOT = os.path.dirname(lightning.app._PROJECT_ROOT)
+_PACKAGE_ROOT = os.path.dirname(__file__)
+_PROJECT_ROOT = os.path.dirname(os.path.dirname(_PACKAGE_ROOT))
```
Comment (carmocca) [inline src/lightning/__init__.py:35]: Why are these 2 variables added? It doesn't seem like anything is using them?

**Review by carmocca — COMMENTED** [review]
Can you explain how these changes fix the problem?

File: .github/workflows/ci-tests-store.yml (hunk 1 above)
Comment (Borda) [reply .github/workflows/ci-tests-store.yml:17]: They are taken from checkgroup :)

File: src/lightning/__init__.py (hunk 2 above)
Comment (Borda) [reply src/lightning/__init__.py:35]:
```suggestion
```

**Comment by Borda** [comment]
> Can you explain how these changes fix the problem?

because it imports something you do not need

**Review by ethanwharris — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Pull request merged**